searchlib/src/vespa/searchlib/transactionlog/domain.cpp

```diff
@@ -354,16 +354,17 @@ int Domain::closeSession(int sessionId)
         if (found != _sessions.end()) {
             sessionRunTime = (std::chrono::steady_clock::now() - found->second->getStartTime());
             retval = 1;
-            _sessionExecutor.sync();
         }
     }
-    if (retval == 1) {
+    while (retval == 1) {
         FastOS_Thread::Sleep(10);
-        LockGuard guard(_sessionLock);
+        MonitorGuard guard(_sessionLock);
```

Comment:
Why change from LockGuard to MonitorGuard ?

Reply:
Unintentional - reverted